projects/kit/components/input-number/input-number.component.ts

```diff
@@ -163,6 +163,12 @@ export class TuiInputNumberComponent
     }
 
     get calculatedMaxLength(): number {
+        const nativeMaxLength = this.textfield?.nativeFocusableElement?.maxLength;
+
+        if (nativeMaxLength !== -1 && nativeMaxLength !== undefined) {
```

Comment:
[Unit tests / kit](https://github.com/taiga-family/taiga-ui/actions/runs/5871927108/job/15922430479?pr=5119) job has failed with the following error:

```console
 FAIL  projects/kit/components/input-number/test/input-number.component.spec.ts
  ● InputNumber › maxlength is set to 23 by default (18 digits + 5 default separators)

    expect(received).toBe(expected) // Object.is equality

    Expected: "23"
    Received: "524288"

      335 |         const nativeInput = getNativeInput()!.nativeElement;
      336 |
    > 337 |         expect(nativeInput.getAttribute(`maxlength`)).toBe(`23`);
          |                                                       ^
      338 |     });
      339 |
      340 |     describe(`When decimal === always`, () => {

      at projects/kit/components/input-number/test/input-number.component.spec.ts:337:55
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:409:30)
      at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:3830:43)
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:408:56)
      at Zone.Object.<anonymous>.Zone.run (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:169:47)
      at Object.wrappedFunc (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:4330:34)
```